li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/metadataservice/MetadataServiceClient.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License").  See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.clients.metadataservice;
+
+import com.linkedin.kafka.clients.common.ClusterGroupDescriptor;
+
+import java.io.Closeable;
+import java.util.Map;
+import java.util.UUID;
+
+import org.apache.kafka.common.Configurable;
+
+
+// An interface to a generic metadata service which serves cluster and topic metadata for federated Kafka clusters.
+public interface MetadataServiceClient extends Configurable, Closeable {
+  /**
+   * Register a federated client with the metadata service. Called by the client to be registered.
+   *
+   * @param clusterGroup  The cluster group descriptor
+   * @param configs  Client configs
+   * @return The id of the registered client
+   */
+  public UUID registerFederatedClient(ClusterGroupDescriptor clusterGroup, Map<String, ?> configs);
```

Comment:
ideally you'd have more things passed to registration - client version, client location (im being vague in purpose, but think hostname)

Reply:
Shouldn't such information be private to a specific mds client implementation? And information that can be passed from a federated client to the corresponding mds client can be passed via configs.

Reply:
needs more thought, but can be done later, so this is good enough for now. the short answer however is that some things (like version of li-apache-kafka-clients, version of kafka etc) should be common across all implementations

Reply:
sgtm